packages/components/src/components/CollapsibleBox/CollapsibleBox.tsx

```diff
@@ -57,270 +64,185 @@ type HeaderProps = {
-};
-
-const Header = styled.div<HeaderProps>`
-    display: flex;
-    justify-content: space-between;
-    align-items: center;
-    gap: ${spacingsPx.xl};
-    padding: ${({ $paddingType }) => headerPaddingTypeMap[$paddingType]};
+const Header = styled.header<HeaderProps>`
+    padding: ${mapPaddingTypeToHeaderPadding};
     cursor: pointer;
 
     &:hover {
-        ${IconWrapper} {
+        ${Toggle} {
             opacity: 0.5;
         }
     }
 `;
 
-const IconLabel = styled.div`
-    margin-right: ${spacingsPx.sm};
-    color: ${({ theme }) => theme.textSubdued};
-    ${typography.hint}
-`;
-
-const Heading = styled.span<HeaderProps>`
-    display: flex;
-    align-items: center;
-    ${typography.body}
-`;
-
-const SubHeading = styled.span`
-    ${typography.hint}
-    color: ${({ theme }) => theme.textSubdued};
-`;
-
-const Flex = styled.div`
-    flex: 1;
-`;
-
-const easingValues = motionEasing.transition.join(', ');
-const StyledIcon = styled(Icon)<{ $isCollapsed?: boolean }>`
+const IconWrapper = styled.div<{ $isCollapsed?: boolean }>`
     transform: ${({ $isCollapsed }) => ($isCollapsed ? 'rotate(0deg)' : 'rotate(180deg)')};
-
-    /* to sync with the expand animation */
-    transition: transform ${ANIMATION_DURATION}s cubic-bezier(${easingValues});
+    transition: transform ${ANIMATION_DURATION}s cubic-bezier(${motionEasing.transition.join(', ')});
     transform-origin: center;
 `;
 
-const contentPaddingTypeMap: Record<PaddingType, string> = {
-    none: '0',
-    normal: `${spacingsPx.lg} ${spacingsPx.md}`,
-    large: `${spacingsPx.xl} ${spacingsPx.md}`,
-};
-
 const Content = styled.div<ContentProps>`
     display: flex;
     flex-direction: column;
-    padding: ${({ $paddingType }) => contentPaddingTypeMap[$paddingType]};
+    padding: ${mapPaddingTypeToContentPadding};
 
-    ${({ $fillType, theme, $elevation, $hasDivider }) =>
-        $fillType &&
+    ${({ theme, $elevation, $hasDivider }) =>
         $hasDivider &&
         css`
             border-top: 1px solid ${mapElevationToBorder({ $elevation, theme })};
         `}
+
+    ${({ $paddingType, $hasDivider }) => css`
+        ${$paddingType === 'none' && $hasDivider && `margin-top: ${spacingsPx.xs};`}
+        ${$paddingType !== 'none' && !$hasDivider && `padding-top: 0;`}
+    `}
 `;
 
 const Collapser = styled(motion.div)`
     overflow: hidden;
 `;
 
-const CollapsibleBoxContent = ({
-    isOpen,
-    onToggle,
-    iconLabel,
+export const CollapsibleBox = ({
+    isOpen = false,
+    toggleLabel,
+    toggleComponent,
     paddingType = 'normal',
     heading,
     subHeading,
-    isIconFlipped = false,
+    headingSize = 'large',
     fillType = 'default',
     hasDivider = true,
     children,
-    className,
     onAnimationComplete,
     'data-testid': dataTest,
     ...rest
-}: CollapsibleBoxContentProps) => {
+}: CollapsibleBoxProps) => {
     const { elevation } = useElevation();
+    const [isActive, setIsActive] = useState(isOpen);
```

Comment:
Would it be better to rename `isOpen` to `defaultIsOpen` and rename `isActive` to `isOpen`?

Also, don't we want to add also managed state?

Reply:
We don't have a use case for that yet, so I'd leave it for now. Agree with the naming suggestion.